### DISCUSSION TITLE
Upload endpoint returns 403

### DISCUSSION THREAD

**Discussion by tilschuenemann** [saleweaver/python-amazon-sp-api#1116] [FAQ]
**Describe the bug**
When trying to upload an image file using the Upload endpoint, I get an error:

```
sp_api.base.exceptions.SellingApiForbiddenException: [{'code': 'Unauthorized', 'message': 'Access to requested resource is denied.', 'details': ''}]
```

**To Reproduce**

I'm running this script:

```python
from sp_api.api import Upload
from sp_api.base import Marketplaces
from PIL import Image

credentials = dict(
    lwa_app_id="xxx",
    lwa_client_secret="xxx",
    aws_access_key="xxx",
    aws_secret_key="xxx",
    refresh_token="xxx"
)

u = Upload(
    marketplace=Marketplaces.ES,
    credentials=credentials,
)
res = u.upload_document(resource="/aplus/2020-11-01/contentDocuments", file=image_path, content_type="image/png")
```

The exact error message:

```python
Traceback (most recent call last):
  File "/home/til/code/monorepo/src/mpm/mahler.py", line 714, in <module>
    main()
  File "/home/til/code/monorepo/src/mpm/mahler.py", line 701, in main
    m.acm_es.upl.upload_document(resource="/uploads/v1/uploadDestinations/aplus/2020-11-01/contentDocuments", file="playground/test.png")
  File "/home/til/.local/lib/python3.11/site-packages/sp_api/base/helpers.py", line 21, in wrapper
    return function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/til/.local/lib/python3.11/site-packages/sp_api/api/upload/upload.py", line 15, in upload_document
    return self._request(fill_query_params(kwargs.pop('path'), resource), params=kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/til/.local/lib/python3.11/site-packages/sp_api/base/client.py", line 157, in _request
    return self._check_response(res, res_no_data, bulk, wrap_list)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/til/.local/lib/python3.11/site-packages/sp_api/base/client.py", line 183, in _check_response
    raise exception(error, headers=res.headers)
sp_api.base.exceptions.SellingApiForbiddenException: [{'code': 'Unauthorized', 'message': 'Access to requested resource is denied.', 'details': ''}]

```

**Expected behavior**

No errors + successful upload.

**Desktop (please complete the following information):**
Arch Linux, Kernel 6.1.38-2 LTS
Python 3.11.3 64 bit

**Additional context**
I'm running a fresh setup of my policies, roles, and user. All token are valid and work, eg. when using the AplusContent endpoint. Most importantly I have the Product Listing role.

I reference the endpoint used in the [example docs by Amazon](https://developer-docs.amazon.com/sp-api/docs/a-content-api-use-case-guide#step-2-create-a-content-document).

Are there any other permissions I need?

**Comment by saleweaver** [comment]
Converted to discussion since it’s unrelated to the library.

**Reply by tilschuenemann** [reply]
I'd usually test both methods _createUploadDestinationForResource_ and upload my file to that destination, but my impression is that the library manages both of these steps and therefore this issue is related to the library?

**Reply by tilschuenemann** [reply]
@saleweaver Can you comment on this or does this lean too much into consulting? We'd be happy to get closure on this.

**Reply by bbenda02** [reply]
Yeah, can we get some clarification on this? I'm stuck up on this issue as well.

**Reply by bbenda02** [reply]
Have you figured anything out on this issue?

**Comment by marios-keri** [comment]
the resource should be "aplus/2020-11-01/contentDocuments" and not  "/aplus/2020-11-01/contentDocuments" without the backslash in the begining.

**Comment by Zengyinqi** [comment]
I also made the same mistake. How did you solve it

**Comment by Zengyinqi** [comment]
I also made the same mistake. How did you solve it

**Comment by Zengyinqi** [comment]
I also made the same mistake. How did you solve it

**Comment by Zengyinqi** [comment]
I also made the same mistake. How did you solve it
